### PR TITLE
Update python dependencies and lock them

### PR DESCRIPTION
**Pull request by gytsto**
### Problem
*--describe problem being solved--*

### Solution
*--describe selected solution--*

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -7,13 +7,13 @@ libtelio-build-pipeline:
...
+    # Don't checkout libtelio submodule when using GIT_SUBMODULE_STRATEGY.
     # This ensures that all jobs that use `libtelio` submodule are calling
     # `ci/checkout.sh`, which uses `LIBTELIO_COMMIT_SHA` to
     # checkout the correct libtelio REF.
     GIT_SUBMODULE_PATHS: ':(exclude)libtelio'
 
   trigger:
     project: $LIBTELIO_BUILD_PROJECT_PATH
-    branch: v0.4.10
+    branch: LLT-4636_update_python_dependencies
```
Comment (Jauler) [inline .gitlab-ci.yml:18]: Adding a reminder message to change this into semver, once above gets merged.

**Comment by Jauler** [comment]
Should'nt all `pip3 install` invocations be changed to pipenv in order for `Pipfile` to get used?

**Comment by gytsto** [comment]
> Should'nt all `pip3 install` invocations be changed to pipenv in order for `Pipfile` to get used?
> 
> E.g. like in `.github/workflows/linters.yml` or similar

not all of them, but the exact example you gave, proably should

**Review by Jauler — APPROVED** [review]
+2.0

**Pull request merged**